src/Resources/skeleton/forgottenPassword/PasswordResetToken.tpl.php

```diff
@@ -0,0 +1,87 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace ?>;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class <?= $class_name ?>
+
+{
+    const TTL_HOURS = 24;
+    const SELECTOR_LENGTH = 20; // in chars
+
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $selector;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $token;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $requestedAt;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="<?= $user_full_class_name ?>")
+     */
+    private $user;
+
+    private $plainToken;
+
+    public function __construct(User $user)
+    {
+        $this->requestedAt = new \DateTimeImmutable('now');
+        $this->selector = strtr(base64_encode(random_bytes(self::SELECTOR_LENGTH * 3 / 4)), '+/', '-_');
+        $this->plainToken = strtr(base64_encode(random_bytes(18)), '+/', '-_');
+        $this->token = password_hash($this->plainToken, PASSWORD_BCRYPT);
```

Comment:
Can we use `PASSWORD_DEFAULT` here? I think that means it would continue improving over time as the app's php versions, etc, increased. Or could we use the new `NativePasswordEncoder` class? I would be ok requiring Symfony 4.3 for this feature if it means we're able to eliminate sensitive code (things that generate and check password) from the generated code

Reply:
I changed to `PASSWORD_DEFAULT`. I think moving to `NativePasswordEncoder` should be done later, at least after 4.4 LTS release. Right now it sounds too much of a restriction, Symfony 4.3 is relatively new.